source/prop.js

```diff
@@ -24,5 +25,10 @@ import path from './path';
  *      R.compose(R.inc, R.prop('x'))({ x: 3 }) //=> 4
  */
 
-var prop = _curry2(function prop(p, obj) { return path([p], obj); });
+var prop = _curry2(function prop(p, obj) {
+  if (obj == null) {
+    return;
+  }
+  return _isInteger(p) ? nth(p, obj) : obj[p];
```

Comment:
Wouldn't `obj[p]` work for both instances? Is there a case where nth would return something different?

Reply:
This is the code copied from "paths". I didn't want to change any logic. Add tests to this case?

Reply:
Interesting. I don't know the history behind this code. For both cases I would think we don't need the check, however I wouldn't want to block the PR on that. I am OK leaving it as-is.

Reply:
`nth` does support negative indices, which normal array access does not: 

```js
const fruit = ['banana', 'apple', 'peach'];

[
  fruit[-1], 
  nth(-1, fruit),
  fruit[-1] === nth(-1, fruit),
]
```
[REPL evaluates it to `[undefined, "peach", false]`](https://ramdajs.com/repl/#?const%20fruit%20%3D%20%5B%27banana%27%2C%20%27apple%27%2C%20%27peach%27%5D%3B%0A%0A%5B%0A%20%20fruit%5B-1%5D%2C%20%0A%20%20nth%28-1%2C%20fruit%29%2C%0A%20%20fruit%5B-1%5D%20%3D%3D%3D%20nth%28-1%2C%20fruit%29%2C%0A%5D)

Reply:
We will need to write tests for this logic. And review rest of code. Becasue, for example, `propIs` doesn't have `nth()` inside...

Reply:
Support for negative indices keeps causing issues!